src/Neo/Persistence/DataCache.cs

```diff
@@ -202,13 +202,37 @@ public void Delete(StorageKey key)
         /// Finds the entries starting with the specified prefix.
         /// </summary>
         /// <param name="key_prefix">The prefix of the key.</param>
+        /// <param name="direction">The search direction.</param>
         /// <returns>The entries found with the desired prefix.</returns>
-        public IEnumerable<(StorageKey Key, StorageItem Value)> Find(byte[] key_prefix = null)
+        public IEnumerable<(StorageKey Key, StorageItem Value)> Find(byte[] key_prefix = null, SeekDirection direction = SeekDirection.Forward)
         {
-            foreach (var (key, value) in Seek(key_prefix, SeekDirection.Forward))
+            var seek_prefix = key_prefix;
+            if (direction == SeekDirection.Backward)
+            {
+                if (key_prefix == null || key_prefix.Length == 0)
+                { // Backwards seek for zero prefix is not supported for now.
+                    throw new ArgumentException();
+                }
+                seek_prefix = null;
+                for (int i = key_prefix.Length - 1; i >= 0; i--)
+                {
+                    if (key_prefix[i] < 0xff)
+                    {
+                        seek_prefix = key_prefix.Take(i + 1).ToArray();
+                        // The next key after the key_prefix.
+                        seek_prefix[i]++;
+                        break;
+                    }
+                }
```

Comment:
Why the seek prefix it's different if is `Backward`?

Reply:
See the example in UT, suppose we have "jjj", "key1", "key2" and "key3" items in the DB and we want to search for items with the "key" prefix. So to search in forward direction we place the cursor at the "key" item (nonexistent, but anyway) and move forward, retreiving items we want:
```
jjj
     <- cursor at "key"
key1
key2
key3
```
If you're to just move in the opposite direction you'll get a `jjj` item (or nothing if we're to check for the prefix additionally. So we do something like this here:
```
jjj
key1
key2
key3
     <- cursor at "keyz"
```
And get "key3", "key2" and "key1".